### ISSUE TITLE
Salt-minion crashed when "Attempt to authenticate with the salt master failed"

### ISSUE THREAD

**Issue by abcfy2** [saltstack/salt#19861]
Hi all,

My OS is CentOS 6.5. I find an issue that salt-minion often crashed when "Attempt to authenticate with the salt master failed".

```
#On minion
/etc/init.d/salt-minion status
salt-minion dead but pid file exists

tail /var/log/salt/minion
2015-01-20 15:16:54,337 [salt                                     ][ERROR   ] Attempt to authenticate with the salt master failed

salt-minion --versions
           Salt: 2014.7.0
         Python: 2.6.6 (r266:84292, Jan 22 2014, 09:42:36)
         Jinja2: unknown
       M2Crypto: 0.20.2
 msgpack-python: 0.1.13
   msgpack-pure: Not Installed
       pycrypto: 2.0.1
        libnacl: Not Installed
         PyYAML: 3.10
          ioflo: Not Installed
          PyZMQ: 14.3.1
           RAET: Not Installed
            ZMQ: 3.2.4
           Mako: 0.3.4
```

```
#On master
 salt --versions
           Salt: 2014.7.0
         Python: 2.6.6 (r266:84292, Jan 22 2014, 09:42:36)
         Jinja2: 2.2.1
       M2Crypto: 0.20.2
 msgpack-python: 0.1.13
   msgpack-pure: Not Installed
       pycrypto: 2.0.1
        libnacl: Not Installed
         PyYAML: 3.09
          ioflo: Not Installed
          PyZMQ: 14.3.1
           RAET: Not Installed
            ZMQ: 3.2.4
           Mako: 0.3.4
```

**Comment by basepi** [comment]
Can we see some debug logs on the master during the authentication failure? I'm thinking there may be more information there. It might also be useful to delete the minion's key from the master and then re-accept the key, there may be some corruption or something there.

**Comment by abcfy2** [comment]
OK. I've already set my salt-minion ``log_level`` to ``debug``. Waiting for the next crash.

**Comment by morsik** [comment]
We have the same problem.

**Comment by abcfy2** [comment]
Hi @basepi , now I found salt-minion has been crashed.

Here is the last log. Please check. Thank you.

```
2015-02-11 08:35:35,128 [salt.crypt                                  ][DEBUG   ] Decrypting the current master AES key
2015-02-11 08:35:35,129 [salt.crypt                                  ][DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
2015-02-11 08:35:35,244 [salt.crypt                                  ][DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
2015-02-11 08:35:35,508 [salt.crypt                                  ][DEBUG   ] Decrypting the current master AES key
2015-02-11 08:35:35,509 [salt.crypt                                  ][DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
2015-02-11 09:35:34,212 [salt.utils.schedule                         ][INFO    ] Running scheduled job: __mine_interval
2015-02-11 09:35:34,212 [salt.utils.schedule                         ][DEBUG   ] schedule: This job was scheduled with jid_include, adding to cache (jid_include defaults to True)
2015-02-11 09:35:34,212 [salt.utils.schedule                         ][DEBUG   ] schedule: This job was scheduled with a max number of 2
2015-02-11 09:35:34,221 [salt.utils.schedule                         ][DEBUG   ] schedule.handle_func: adding this job to the jobcache with data {'fun': 'mine.update', 'jid': '20150211093534215008', 'pid': 24789, 'id': 'ip-112.1.30.3-XuYiServer', 'schedule': '__mine_interval'}
2015-02-11 09:35:34,683 [salt.crypt                                  ][DEBUG   ] Decrypting the current master AES key
2015-02-11 09:35:34,684 [salt.crypt                                  ][DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
2015-02-11 09:35:34,799 [salt.crypt                                  ][DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
2015-02-11 09:35:38,814 [salt.crypt                                  ][DEBUG   ] Decrypting the current master AES key
2015-02-11 09:35:38,815 [salt.crypt                                  ][DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
2015-02-11 10:35:34,707 [salt.utils.schedule                         ][INFO    ] Running scheduled job: __mine_interval
2015-02-11 10:35:34,707 [salt.utils.schedule                         ][DEBUG   ] schedule: This job was scheduled with jid_include, adding to cache (jid_include defaults to True)
2015-02-11 10:35:34,707 [salt.utils.schedule                         ][DEBUG   ] schedule: This job was scheduled with a max number of 2
2015-02-11 10:35:34,715 [salt.utils.schedule                         ][DEBUG   ] schedule.handle_func: adding this job to the jobcache with data {'fun': 'mine.update', 'jid': '20150211103534710077', 'pid': 26257, 'id': 'ip-112.1.30.3-XuYiServer', 'schedule': '__mine_interval'}
2015-02-11 10:35:34,980 [salt.crypt                                  ][DEBUG   ] Decrypting the current master AES key
2015-02-11 10:35:34,980 [salt.crypt                                  ][DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
2015-02-11 10:35:35,096 [salt.crypt                                  ][DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
2015-02-11 10:35:35,664 [salt.crypt                                  ][DEBUG   ] Decrypting the current master AES key
2015-02-11 10:35:35,664 [salt.crypt                                  ][DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
2015-02-11 11:35:34,894 [salt.utils.schedule                         ][INFO    ] Running scheduled job: __mine_interval
2015-02-11 11:35:34,896 [salt.utils.schedule                         ][DEBUG   ] schedule: This job was scheduled with jid_include, adding to cache (jid_include defaults to True)
2015-02-11 11:35:34,896 [salt.utils.schedule                         ][DEBUG   ] schedule: This job was scheduled with a max number of 2
2015-02-11 11:35:34,920 [salt.utils.schedule                         ][DEBUG   ] schedule.handle_func: adding this job to the jobcache with data {'fun': 'mine.update', 'jid': '20150211113534905429', 'pid': 28882, 'id': 'ip-112.1.30.3-XuYiServer', 'schedule': '__mine_interval'}
2015-02-11 11:35:35,167 [salt.crypt                                  ][DEBUG   ] Decrypting the current master AES key
2015-02-11 11:35:35,167 [salt.crypt                                  ][DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
2015-02-11 11:35:35,388 [salt.crypt                                  ][DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
2015-02-11 11:35:35,733 [salt.crypt                                  ][DEBUG   ] Decrypting the current master AES key
2015-02-11 11:35:35,734 [salt.crypt                                  ][DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
2015-02-11 12:35:34,645 [salt.utils.schedule                         ][INFO    ] Running scheduled job: __mine_interval
2015-02-11 12:35:34,646 [salt.utils.schedule                         ][DEBUG   ] schedule: This job was scheduled with jid_include, adding to cache (jid_include defaults to True)
2015-02-11 12:35:34,647 [salt.utils.schedule                         ][DEBUG   ] schedule: This job was scheduled with a max number of 2
2015-02-11 12:35:34,671 [salt.utils.schedule                         ][DEBUG   ] schedule.handle_func: adding this job to the jobcache with data {'fun': 'mine.update', 'jid': '20150211123534658886', 'pid': 29753, 'id': 'ip-112.1.30.3-XuYiServer', 'schedule': '__mine_interval'}
2015-02-11 12:35:34,905 [salt.crypt                                  ][DEBUG   ] Decrypting the current master AES key
2015-02-11 12:35:34,906 [salt.crypt                                  ][DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
2015-02-11 12:35:35,076 [salt.crypt                                  ][DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
2015-02-11 12:35:35,419 [salt.crypt                                  ][DEBUG   ] Decrypting the current master AES key
2015-02-11 12:35:35,420 [salt.crypt                                  ][DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
2015-02-11 13:35:34,323 [salt.utils.schedule                         ][INFO    ] Running scheduled job: __mine_interval
2015-02-11 13:35:34,325 [salt.utils.schedule                         ][DEBUG   ] schedule: This job was scheduled with jid_include, adding to cache (jid_include defaults to True)
2015-02-11 13:35:34,325 [salt.utils.schedule                         ][DEBUG   ] schedule: This job was scheduled with a max number of 2
2015-02-11 13:35:34,357 [salt.utils.schedule                         ][DEBUG   ] schedule.handle_func: adding this job to the jobcache with data {'fun': 'mine.update', 'jid': '20150211133534337222', 'pid': 30915, 'id': 'ip-112.1.30.3-XuYiServer', 'schedule': '__mine_interval'}
2015-02-11 13:35:35,589 [salt.crypt                                  ][DEBUG   ] Decrypting the current master AES key
2015-02-11 13:35:35,589 [salt.crypt                                  ][DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
2015-02-11 13:35:35,824 [salt.crypt                                  ][DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
2015-02-11 13:35:36,061 [salt.crypt                                  ][DEBUG   ] Decrypting the current master AES key
2015-02-11 13:35:36,062 [salt.crypt                                  ][DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
2015-02-11 14:35:34,697 [salt.utils.schedule                         ][INFO    ] Running scheduled job: __mine_interval
2015-02-11 14:35:34,701 [salt.utils.schedule                         ][DEBUG   ] schedule: This job was scheduled with jid_include, adding to cache (jid_include defaults to True)
2015-02-11 14:35:34,701 [salt.utils.schedule                         ][DEBUG   ] schedule: This job was scheduled with a max number of 2
2015-02-11 14:35:34,717 [salt.utils.schedule                         ][DEBUG   ] schedule.handle_func: adding this job to the jobcache with data {'fun': 'mine.update', 'jid': '20150211143534711429', 'pid': 473, 'id': 'ip-112.1.30.3-XuYiServer', 'schedule': '__mine_interval'}
2015-02-11 14:35:34,908 [salt.crypt                                  ][DEBUG   ] Decrypting the current master AES key
2015-02-11 14:35:34,909 [salt.crypt                                  ][DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
2015-02-11 14:35:35,023 [salt.crypt                                  ][DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
2015-02-11 14:35:35,509 [salt.crypt                                  ][DEBUG   ] Decrypting the current master AES key
2015-02-11 14:35:35,509 [salt.crypt                                  ][DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
2015-02-11 15:35:34,532 [salt.utils.schedule                         ][INFO    ] Running scheduled job: __mine_interval
2015-02-11 15:35:34,532 [salt.utils.schedule                         ][DEBUG   ] schedule: This job was scheduled with jid_include, adding to cache (jid_include defaults to True)
2015-02-11 15:35:34,532 [salt.utils.schedule                         ][DEBUG   ] schedule: This job was scheduled with a max number of 2
2015-02-11 15:35:34,539 [salt.utils.schedule                         ][DEBUG   ] schedule.handle_func: adding this job to the jobcache with data {'fun': 'mine.update', 'jid': '20150211153534534887', 'pid': 2360, 'id': 'ip-112.1.30.3-XuYiServer', 'schedule': '__mine_interval'}
2015-02-11 15:35:34,741 [salt.crypt                                  ][DEBUG   ] Decrypting the current master AES key
2015-02-11 15:35:34,742 [salt.crypt                                  ][DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
2015-02-11 15:35:34,857 [salt.crypt                                  ][DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
2015-02-11 15:35:35,315 [salt.crypt                                  ][DEBUG   ] Decrypting the current master AES key
2015-02-11 15:35:35,316 [salt.crypt                                  ][DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
2015-02-11 18:11:16,164 [salt             ][INFO    ] Setting up the Salt Minion "ip-112.1.30.3-XuYiServer"
2015-02-11 18:11:16,205 [salt.utils.process][DEBUG   ] Created pidfile: /var/run/salt-minion.pid
2015-02-11 18:11:16,207 [salt.config      ][DEBUG   ] Reading configuration from /etc/salt/minion
2015-02-11 18:11:16,883 [salt.minion                              ][DEBUG   ] Attempting to authenticate with the Salt Master at 218.94.143.226
2015-02-11 18:11:17,144 [salt.crypt                               ][DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
2015-02-11 18:12:17,307 [salt                                     ][ERROR   ] Attempt to authenticate with the salt master failed
```

**Comment by basepi** [comment]
I actually wanted the debug logs for the master -- any chance you can get those?

**Comment by abcfy2** [comment]
Sorry, the log level of the master is default warning. I just change it to debug. Waiting for the next.

**Comment by br0ch0n** [comment]
I've just had this happen in my test env.  In my case the salt master wasn't even started (i.e. can't be the master's fault), which is what I first thought the 
```
2015-02-22 21:18:25,204 [salt             ][ERROR   ] Attempt to authenticate with the salt master failed
```
was.  However, after I started up my salt master node and didn't see the minion node connect, I became concerned.  Turns out the minion proc on the minion node decided to crash rather than keep attempting to connect to the (at the time) non-existent master.  Restarted the minion proc and everything's fine of course, but I imagine this is not normal behavior.

**Comment by pipozzz** [comment]
same problem

**Comment by abcfy2** [comment]
I find some logs.

```
  File "/usr/lib/python2.7/dist-packages/salt/modules/mine.py", line 25, in _auth
    __context__['auth'] = salt.crypt.SAuth(__opts__)
  File "/usr/lib/python2.7/dist-packages/salt/crypt.py", line 810, in __init__
    self.crypticle = self.__authenticate()
  File "/usr/lib/python2.7/dist-packages/salt/crypt.py", line 831, in __authenticate
    self.opts.get('auth_tries', 1)
  File "/usr/lib/python2.7/dist-packages/salt/crypt.py", line 649, in sign_in
    raise SaltClientError('Attempt to authenticate with the salt master failed')
SaltClientError: Attempt to authenticate with the salt master failed
2015-03-09 15:21:19,264 [salt             ][ERROR   ] Attempt to authenticate with the salt master failed
  File "/usr/lib/python2.7/dist-packages/salt/modules/mine.py", line 25, in _auth
    __context__['auth'] = salt.crypt.SAuth(__opts__)
  File "/usr/lib/python2.7/dist-packages/salt/crypt.py", line 810, in __init__
    self.crypticle = self.__authenticate()
  File "/usr/lib/python2.7/dist-packages/salt/crypt.py", line 831, in __authenticate
    self.opts.get('auth_tries', 1)
  File "/usr/lib/python2.7/dist-packages/salt/crypt.py", line 649, in sign_in
    raise SaltClientError('Attempt to authenticate with the salt master failed')
SaltClientError: Attempt to authenticate with the salt master failed
```

**Comment by basepi** [comment]
I think `auth_safemode` only applies to `SaltReqTimeoutError`s so hopefully we'll be able to just extend that logic to these SaltClientErrors as well.

**Comment by chenqing** [comment]
The same problem to me,  today the network is break between master and mioin ,and salt version is : 
salt --version
salt 2014.7.0 (Helium)

os release : Red Hat Enterprise Linux Server release 5.8 (Tikanga)

and  error log : 


2015-03-27 04:04:34,741 [salt.utils.schedule                         ][ERROR   ] Unhandled exception running mine.update
Traceback (most recent call last):
  File "/home/salt/python2.7/lib/python2.7/site-packages/salt/utils/schedule.py", line 421, in handle_func
    ret['return'] = self.functions[func]()
  File "/home/salt/python2.7/lib/python2.7/site-packages/salt/modules/mine.py", line 99, in update
    load['tok'] = _auth().gen_token('salt')
  File "/home/salt/python2.7/lib/python2.7/site-packages/salt/modules/mine.py", line 25, in _auth
    __context__['auth'] = salt.crypt.SAuth(__opts__)
  File "/home/salt/python2.7/lib/python2.7/site-packages/salt/crypt.py", line 805, in __init__
    self.crypticle = self.__authenticate()
  File "/home/salt/python2.7/lib/python2.7/site-packages/salt/crypt.py", line 826, in __authenticate
    self.opts.get('auth_tries', 1)
  File "/home/salt/python2.7/lib/python2.7/site-packages/salt/crypt.py", line 644, in sign_in
    raise SaltClientError('Attempt to authenticate with the salt master failed')
SaltClientError: Attempt to authenticate with the salt master failed

**Comment by abcfy2** [comment]
Today this bug has been replicated. But no more logs for debug level.

Execute this command twice:
```
salt "minion" test.ping -v
minion:
    Minion did not return. [No response]
```
On minion:
```
# service salt-minion status
salt-minion stop/waiting
```

minion log:
```
# cat minion 
2015-03-31 15:38:02,871 [salt             ][INFO    ] Setting up the Salt Minion "minion"
2015-03-31 15:38:02,871 [salt.utils.process][DEBUG   ] Created pidfile: /var/run/salt-minion.pid
2015-03-31 15:38:02,871 [salt.config      ][DEBUG   ] Reading configuration from /etc/salt/minion
2015-03-31 15:38:02,993 [salt.minion      ][DEBUG   ] Attempting to authenticate with the Salt Master at 218.94.143.226
2015-03-31 15:38:02,993 [salt.crypt       ][DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
2015-03-31 15:39:03,048 [salt             ][ERROR   ] Attempt to authenticate with the salt master failed
```

master log:
```
2015-03-31 15:43:04,756 [salt.master                                ][INFO    ] Clear payload received with command publish
2015-03-31 15:43:04,757 [salt.loader                                ][DEBUG   ] LazyLoaded local_cache.prep_jid
2015-03-31 15:43:04,758 [salt.utils.event                           ][DEBUG   ] Sending event - data = {'_stamp': '2015-03-31T15:43:04.758236', 'minions': ['minion']}
2015-03-31 15:43:04,759 [salt.utils.event                           ][DEBUG   ] Sending event - data = {'tgt_type': 'glob', 'jid': '20150331154304757534', 'tgt': 'minion', '_stamp': '2015-03-31T15:43:04.758610', 'user': 'root', 'arg': [], 'fun': 'test.ping', 'minions': ['minion']}
2015-03-31 15:43:04,759 [salt.utils.event                           ][DEBUG   ] Sending event - data = {'tgt_type': 'glob', 'jid': '20150331154304757534', 'tgt': 'minion', '_stamp': '2015-03-31T15:43:04.758917', 'user': 'root', 'arg': [], 'fun': 'test.ping', 'minions': ['minion']}
2015-03-31 15:43:04,760 [salt.master                                ][INFO    ] User root Published command test.ping with jid 20150331154304757534
2015-03-31 15:43:04,760 [salt.master                                ][DEBUG   ] Published command details {'tgt_type': 'glob', 'jid': '20150331154304757534', 'tgt': 'minion', 'ret': '', 'user': 'root', 'arg': [], 'fun': 'test.ping'}
2015-03-31 15:43:09,773 [salt.master                                ][INFO    ] Clear payload received with command publish
2015-03-31 15:43:09,774 [salt.loader                                ][DEBUG   ] LazyLoaded local_cache.prep_jid
2015-03-31 15:43:09,775 [salt.utils.event                           ][DEBUG   ] Sending event - data = {'_stamp': '2015-03-31T15:43:09.775077', 'minions': ['minion']}
2015-03-31 15:43:09,776 [salt.utils.event                           ][DEBUG   ] Sending event - data = {'tgt_type': 'glob', 'jid': '20150331154309774267', 'tgt': 'minion', '_stamp': '2015-03-31T15:43:09.775538', 'user': 'root', 'arg': ['20150331154304757534'], 'fun': 'saltutil.find_job', 'minions': ['minion']}
2015-03-31 15:43:09,776 [salt.utils.event                           ][DEBUG   ] Sending event - data = {'tgt_type': 'glob', 'jid': '20150331154309774267', 'tgt': 'minion', '_stamp': '2015-03-31T15:43:09.775912', 'user': 'root', 'arg': ['20150331154304757534'], 'fun': 'saltutil.find_job', 'minions': ['minion']}
2015-03-31 15:43:09,777 [salt.master                                ][INFO    ] User root Published command saltutil.find_job with jid 20150331154309774267
2015-03-31 15:43:09,777 [salt.master                                ][DEBUG   ] Published command details {'tgt_type': 'glob', 'jid': '20150331154309774267', 'tgt': 'minion', 'ret': '', 'user': 'root', 'arg': ['20150331154304757534'], 'fun': 'saltutil.find_job'}
2015-03-31 15:43:27,960 [salt.config                                 ][DEBUG   ] Reading configuration from /etc/salt/master
2015-03-31 15:43:28,062 [salt.config                                 ][DEBUG   ] Guessing ID. The id can be explicitly in set /etc/salt/minion
2015-03-31 15:43:28,073 [salt.config                                 ][INFO    ] Found minion id from generate_minion_id(): LMSSERVER
2015-03-31 15:43:28,074 [salt.config                                 ][DEBUG   ] Reading configuration from /etc/salt/master
2015-03-31 15:43:28,466 [salt.loader                                 ][DEBUG   ] LazyLoaded local_cache.clean_old_jobs
2015-03-31 15:43:28,473 [salt.fileserver                             ][DEBUG   ] Updating fileserver cache
2015-03-31 15:43:28,483 [salt.utils.verify                           ][DEBUG   ] This salt-master instance has accepted 12 minion keys.
2015-03-31 15:43:57,362 [salt.master                                ][INFO    ] Clear payload received with command publish
2015-03-31 15:43:57,363 [salt.loader                                ][DEBUG   ] LazyLoaded local_cache.prep_jid
2015-03-31 15:43:57,364 [salt.utils.event                           ][DEBUG   ] Sending event - data = {'_stamp': '2015-03-31T15:43:57.363635', 'minions': ['minion']}
2015-03-31 15:43:57,364 [salt.utils.event                           ][DEBUG   ] Sending event - data = {'tgt_type': 'glob', 'jid': '20150331154357362937', 'tgt': 'minion', '_stamp': '2015-03-31T15:43:57.364028', 'user': 'root', 'arg': [], 'fun': 'test.ping', 'minions': ['minion']}
2015-03-31 15:43:57,364 [salt.utils.event                           ][DEBUG   ] Sending event - data = {'tgt_type': 'glob', 'jid': '20150331154357362937', 'tgt': 'minion', '_stamp': '2015-03-31T15:43:57.364340', 'user': 'root', 'arg': [], 'fun': 'test.ping', 'minions': ['minion']}
2015-03-31 15:43:57,365 [salt.master                                ][INFO    ] User root Published command test.ping with jid 20150331154357362937
2015-03-31 15:43:57,365 [salt.master                                ][DEBUG   ] Published command details {'tgt_type': 'glob', 'jid': '20150331154357362937', 'tgt': 'minion', 'ret': '', 'user': 'root', 'arg': [], 'fun': 'test.ping'}
2015-03-31 15:44:02,377 [salt.master                                ][INFO    ] Clear payload received with command publish
2015-03-31 15:44:02,378 [salt.loader                                ][DEBUG   ] LazyLoaded local_cache.prep_jid
2015-03-31 15:44:02,379 [salt.utils.event                           ][DEBUG   ] Sending event - data = {'_stamp': '2015-03-31T15:44:02.378663', 'minions': ['minion']}
2015-03-31 15:44:02,379 [salt.utils.event                           ][DEBUG   ] Sending event - data = {'tgt_type': 'glob', 'jid': '20150331154402377959', 'tgt': 'minion', '_stamp': '2015-03-31T15:44:02.379049', 'user': 'root', 'arg': ['20150331154357362937'], 'fun': 'saltutil.find_job', 'minions': ['minion']}
2015-03-31 15:44:02,379 [salt.utils.event                           ][DEBUG   ] Sending event - data = {'tgt_type': 'glob', 'jid': '20150331154402377959', 'tgt': 'minion', '_stamp': '2015-03-31T15:44:02.379362', 'user': 'root', 'arg': ['20150331154357362937'], 'fun': 'saltutil.find_job', 'minions': ['minion']}
2015-03-31 15:44:02,380 [salt.master                                ][INFO    ] User root Published command saltutil.find_job with jid 20150331154402377959
2015-03-31 15:44:02,380 [salt.master                                ][DEBUG   ] Published command details {'tgt_type': 'glob', 'jid': '20150331154402377959', 'tgt': 'minion', 'ret': '', 'user': 'root', 'arg': ['20150331154357362937'], 'fun': 'saltutil.find_job'}
2015-03-31 15:44:28,544 [salt.config                                 ][DEBUG   ] Reading configuration from /etc/salt/master
2015-03-31 15:44:28,643 [salt.config                                 ][DEBUG   ] Guessing ID. The id can be explicitly in set /etc/salt/minion
2015-03-31 15:44:28,655 [salt.config                                 ][INFO    ] Found minion id from generate_minion_id(): LMSSERVER
2015-03-31 15:44:28,656 [salt.config                                 ][DEBUG   ] Reading configuration from /etc/salt/master
2015-03-31 15:44:29,049 [salt.loader                                 ][DEBUG   ] LazyLoaded local_cache.clean_old_jobs
2015-03-31 15:44:29,056 [salt.fileserver                             ][DEBUG   ] Updating fileserver cache
2015-03-31 15:44:29,066 [salt.utils.verify                           ][DEBUG   ] This salt-master instance has accepted 12 minion keys.
2015-03-31 15:45:29,114 [salt.config                                 ][DEBUG   ] Reading configuration from /etc/salt/master
2015-03-31 15:45:29,206 [salt.config                                 ][DEBUG   ] Guessing ID. The id can be explicitly in set /etc/salt/minion
2015-03-31 15:45:29,218 [salt.config                                 ][INFO    ] Found minion id from generate_minion_id(): LMSSERVER
2015-03-31 15:45:29,219 [salt.config                                 ][DEBUG   ] Reading configuration from /etc/salt/master
2015-03-31 15:45:29,627 [salt.loader                                 ][DEBUG   ] LazyLoaded local_cache.clean_old_jobs
2015-03-31 15:45:29,635 [salt.fileserver                             ][DEBUG   ] Updating fileserver cache
2015-03-31 15:45:29,646 [salt.utils.verify                           ][DEBUG   ] This salt-master instance has accepted 12 minion keys.
```

**Comment by ciandro** [comment]
same problem.. it was actually working for me two weeks ago and now it shows the same behavior as everybody else.

**Comment by ravenblue12** [comment]
Same on CentOS 6.6:

           Salt: 2014.7.5
         Python: 2.6.6 (r266:84292, Jan 22 2014, 09:42:36)
         Jinja2: 2.2.1
       M2Crypto: 0.20.2
 msgpack-python: 0.4.6
   msgpack-pure: Not Installed
       pycrypto: 2.0.1
        libnacl: Not Installed
         PyYAML: 3.10
          ioflo: Not Installed
          PyZMQ: 14.3.1
           RAET: Not Installed
            ZMQ: 3.2.5
           Mako: Not Installed

**Comment by JordanP** [comment]
I face the same issue from time to time. Ubuntu 14.04, salt 2014.7. 

In my case, I deploy the salt master and the salt minion concurrently on different boxes, and this issue may arise when the salt-minion finishes to install before the salt master. But I am not sure because that's not 100% reproducible.

**Comment by dennisjac** [comment]
I'm seeing the same thing. A whole bunch of nodes were shown as down in manage.status and the minion on these nodes was no longer running and the last entry in the log was the authentication failure.

**Comment by ytjohn** [comment]
Version: 2014.7.1 (Helium)
We had the same experience. In our case, this was after a multi-day network maintenance in the lab that broke the connection between the minions and the master. Once network connectivity was restored, we discovered all of our minions were dead, with the minions reporting the authentication issue (during a time with the network down).  

However, after restarting the minions, most came back, but some had the auth failure again. Additional restarts brought the minions back online. As an extra data point, we did validate that ntp was working and everyone had the same time.

It appears to be a result of losing contact with the master. It could be that the master was overloaded and not responding for some nodes again.

**Comment by rsuarezsoto** [comment]
Count me in on this. We have a minion that was down for a couple days due to a electricity outtage, and when it came back it started to show this behaviour. We're running 2014.7.1 in the minion and 2014.7.5 in the master, in case it helps.

**Comment by dennisjac** [comment]
Upgrade to 2015.5. I very much doubt that this is going do be dealt with in 2014.7 anymore.

**Comment by cachedout** [comment]
This looks like it was fixed in 2015.5. It's not likely that we're going to release another 2014.7, but the fix for this should be simple for those wishing to hotfix their 2014.7 installations:

Change:

```
         try:
              load['tok'] = _auth().gen_token('salt')
          except AttributeError:
              log.error('Mine could not authenticate with master. Mine data '
                         'not sent.```
```
to

```
         try:
              load['tok'] = _auth().gen_token('salt')
          except (AttributeError, SaltClientError):
              log.error('Mine could not authenticate with master. Mine data '
                         'not sent.```
```

**Comment by quantonganh** [comment]
@cachedout: I got another error after applying your fix:

```
 The minion function caused an exception: Traceback (most recent call last):
      File "/usr/lib/python2.7/dist-packages/salt/minion.py", line 1020, in _thread_return
        return_data = func(*args, **kwargs)
      File "/usr/lib/python2.7/dist-packages/salt/modules/state.py", line 448, in sls
        st_ = salt.state.HighState(opts, pillar, kwargs.get('__pub_jid'))
      File "/usr/lib/python2.7/dist-packages/salt/state.py", line 2967, in __init__
        self.state = State(self.opts, pillar, jid)
      File "/usr/lib/python2.7/dist-packages/salt/state.py", line 560, in __init__
        self.opts['pillar'] = self._gather_pillar()
      File "/usr/lib/python2.7/dist-packages/salt/state.py", line 578, in _gather_pillar
        self.opts['environment'],
      File "/usr/lib/python2.7/dist-packages/salt/pillar/__init__.py", line 62, in get_pillar
        }.get(opts['file_client'], Pillar)(opts, grains, id_, saltenv, ext)
      File "/usr/lib/python2.7/dist-packages/salt/pillar/__init__.py", line 76, in __init__
        self.sreq = salt.transport.Channel.factory(opts)
      File "/usr/lib/python2.7/dist-packages/salt/transport/__init__.py", line 47, in factory
        return ZeroMQChannel(opts, **kwargs)
      File "/usr/lib/python2.7/dist-packages/salt/transport/__init__.py", line 236, in __init__
        self.auth = salt.crypt.SAuth(opts)
      File "/usr/lib/python2.7/dist-packages/salt/crypt.py", line 811, in __init__
        self.crypticle = self.__authenticate()
      File "/usr/lib/python2.7/dist-packages/salt/crypt.py", line 832, in __authenticate
        self.opts.get('auth_tries', 1)
      File "/usr/lib/python2.7/dist-packages/salt/crypt.py", line 650, in sign_in
        raise SaltClientError('Attempt to authenticate with the salt master failed')
    SaltClientError: Attempt to authenticate with the salt master failed
```

salt-call --versions-report:

```
                  Salt: 2014.7.5
                Python: 2.7.6 (default, Jun 22 2015, 17:58:13)
                Jinja2: 2.7.2
              M2Crypto: 0.21.1
        msgpack-python: 0.3.0
          msgpack-pure: Not Installed
              pycrypto: 2.6.1
               libnacl: Not Installed
                PyYAML: 3.11
                 ioflo: Not Installed
                 PyZMQ: 14.0.1
                  RAET: Not Installed
                   ZMQ: 4.0.4
                  Mako: 0.9.1
 Debian source package: 2014.7.5+ds-1ubuntu1
```
